components/create-collective/NewCreateCollective.js

```diff
@@ -0,0 +1,223 @@
+import CollectiveCategoryPicker from './sections/CollectiveCategoryPicker';
+import ConnectGithub from './sections/ConnectGithub';
+import SignInOrJoinFree from '../SignInOrJoinFree';
+import { withUser } from '../UserProvider';
+
+import { addCreateCollectiveMutationV2 } from '../../lib/graphql/mutations';
+import { getErrorFromGraphqlException } from '../../lib/utils';
+import { Router } from '../../server/pages';
+
+class NewCreateCollective extends Component {
+  static propTypes = {
+    host: PropTypes.object,
+    query: PropTypes.object,
+    LoggedInUser: PropTypes.object, // from withUser
+    refetchLoggedInUser: PropTypes.func.isRequired, // from withUser
+    intl: PropTypes.object.isRequired,
+    createCollectiveV2: PropTypes.func,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      collective: {},
+      result: {},
+      category: null,
+      github: null,
+      form: false,
+      error: null,
+    };
+    this.createCollective = this.createCollective.bind(this);
+    this.messages = defineMessages({
+      'host.apply.title': {
+        id: 'host.apply.title',
+        defaultMessage: 'Apply to create a new {hostname} collective',
+      },
+      'collective.create.title': {
+        id: 'collective.create.title',
+        defaultMessage: 'Create an Open Collective',
+      },
+      'collective.create.description': {
+        id: 'collective.create.description',
+        defaultMessage: 'The place for your community to collect money and share your finance in full transparency.',
+      },
+    });
+
+    this.host = props.host;
+  }
+
+  componentDidMount() {
+    const { query } = this.props;
+    if (query.category === 'opensource' || query.token) {
+      this.setState({ category: 'opensource' });
+      if (query.step === 'form') {
+        this.setState({ form: true });
+      }
+      if (!query.step) {
+        this.setState({ form: false });
+      }
+    } else if (query.category === 'community') {
+      this.setState({ category: 'community' });
+    } else if (query.category === 'climate') {
+      this.setState({ category: 'climate' });
+    } else if (!query.category) {
+      this.setState({ category: null });
+    }
+    return;
+  }
+
+  componentDidUpdate(oldProps) {
+    const { query } = this.props;
+    if (oldProps.query.step !== query.step) {
+      if (query.step === 'form') {
+        this.setState({ form: true });
+      } else {
+        this.setState({ form: false });
+      }
+    }
+    if (oldProps.query.category !== query.category) {
+      if (query.category === 'opensource' || query.token) {
+        this.setState({ category: 'opensource' });
+      } else if (query.category === 'community') {
+        this.setState({ category: 'community' });
+      } else if (query.category === 'climate') {
+        this.setState({ category: 'climate' });
+      } else if (!query.category) {
+        this.setState({ category: null });
+      }
+    }
+    return;
```

Comment:
There's a lot of code like this that seems to be leftovers from the previous create collective page, it makes it pretty difficult to do a proper review. Can we remote it?